frontend/src/lib/components/Subscriptions/views/EditSubscription.tsx

```diff
@@ -63,14 +72,36 @@ export function EditSubscription({
     }
 
     const commonSelectProps: Partial<LemonSelectProps<LemonSelectOptions>> = {
-        dropdownPlacement: 'top-start',
         type: 'stealth',
         outlined: true,
-        disabled: emailDisabled,
-        dropdownMatchSelectWidth: false,
         dropdownMaxWindowDimensions: true,
     }
 
+    useEffect(() => {
+        if (subscription.target_type === 'slack') {
+            loadSlackChannels()
+        }
+    }, [subscription.target_type])
```

Comment:
Could this also be triggered in the logic? It's usually good to keep all the knowledge of how the data changes in the kea layer...